### PR TITLE
Transitions

### PR DESCRIPTION
**Pull request by Jascha-Sundaresan**
This gets us off react-addons-css-transition-group which has been deprecated.
This also removes two other dependencies: react-router-transitions and react-transitions-group.

All that remains is react-motion, which was previously a dependency of react-router-transitions, but is now in our package.json explicitly instead.

I needed to do this to get us back up to react 16. These transitions will also be much easier to convert to react-router-4 when we make that upgrade.  Getting us back to react 16 fixes the regressions in my work on removing react-tooltip.

Thank you

### CODE REVIEW

File: app/components/views/TransactionsPage/Header.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { spring, Motion } from "react-motion";
+import { Link } from "react-router";
+import { injectIntl, defineMessages } from "react-intl";
+import "style/Header.less";
+import headerConnector from "connectors/transactionsPage";
+
+const opts = { stiffness: 150, damping: 15 };
```
Comment (alexlyp) [inline app/components/views/TransactionsPage/Header.js]: I feel like it's a bit bouncy now, after testing a bit `damping: 20` feels good.

File: app/components/views/TransactionsPage/index.js [hunk 2]
```diff
@@ -1,41 +1,28 @@
...
+import Header from "./Header";
+import { RouteTransition } from "shared";
 
-const propTypes = {
-  router: PropTypes.object.isRequired,
-  isTestNet: PropTypes.bool.isRequired,
-};
+const enterLeft = { atEnter: { offset: -100 }, atActive: { offset: 0 }, atLeave: { offset: 100 }, mapStyles: (styles) => ({ transform: `translateX(${styles.offset}%)` })};
+const enterRight = { atEnter: { offset: 100 }, atActive: { offset: 0 }, atLeave: { offset: -100 }, mapStyles: (styles) => ({ transform: `translateX(${styles.offset}%)` })};
+const opts = { stiffness: 150, damping: 15 };
```
Comment (alexlyp) [inline app/components/views/TransactionsPage/index.js]: Same `damping: 20`

File: app/style/Header.less [hunk 3]
```diff
@@ -1,123 +1,114 @@
...
+
+.tabbedheader-tab {
+  display: inline-block;
+  vertical-align: middle;
+  text-align: center;
+  padding: 0px 4px 3px 4px;
+  margin: 0px 10px 0px 10px;
+  box-sizing: border-box;
+  color: inherit;
+  text-decoration: none;
```
Comment (alexlyp) [inline app/style/Header.less]: I see the text-decoration set to none, but looks like the tabs have default `<a>` styling.

**Comment by Jascha-Sundaresan** [comment]
I'm closing this PR and opening another one from a fresh branch w/ a cleaner commit history.

File: app/components/views/TransactionsPage/Header.js (hunk 1 above)
Comment (Jascha-Sundaresan) [reply app/components/views/TransactionsPage/Header.js]: sure.. I got the numbers 150 and 15 from design but I'll change them

File: app/components/views/TransactionsPage/index.js (hunk 2 above)
Comment (Jascha-Sundaresan) [reply app/components/views/TransactionsPage/index.js]: I can set defaults inside RouteTransition if you want.

File: app/style/Header.less (hunk 3 above)
Comment (Jascha-Sundaresan) [reply app/style/Header.less]: this is from before... I didn't add or change any styling rules.

Comment (alexlyp) [reply app/style/Header.less]: Well no matter the cause, it needs to be fixed

**Comment by alexlyp** [comment]
No need to close PRs.  Just do you cleaning then force push to this branch

File: app/components/views/TransactionsPage/index.js (hunk 2 above)
Comment (alexlyp) [reply app/components/views/TransactionsPage/index.js]: sure

**Pull request closed without merging**